Build.bee.cs

```diff
@@ -0,0 +1,154 @@
+                "src/google/protobuf/reflection_ops.cc",
+                "src/google/protobuf/service.cc",
+                "src/google/protobuf/source_context.pb.cc",
+                "src/google/protobuf/struct.pb.cc",
+                "src/google/protobuf/stubs/mathlimits.cc",
+                "src/google/protobuf/stubs/substitute.cc",
+                "src/google/protobuf/text_format.cc",
+                "src/google/protobuf/timestamp.pb.cc",
+                "src/google/protobuf/type.pb.cc",
+                "src/google/protobuf/unknown_field_set.cc",
+                "src/google/protobuf/util/delimited_message_util.cc",
+                "src/google/protobuf/util/field_comparator.cc",
+                "src/google/protobuf/util/field_mask_util.cc",
+                "src/google/protobuf/util/internal/datapiece.cc",
+                "src/google/protobuf/util/internal/default_value_objectwriter.cc",
+                //"src/google/protobuf/util/internal/error_listener.cc", // Empty file
+                "src/google/protobuf/util/internal/field_mask_utility.cc",
+                "src/google/protobuf/util/internal/json_escaping.cc",
+                "src/google/protobuf/util/internal/json_objectwriter.cc",
+                "src/google/protobuf/util/internal/json_stream_parser.cc",
+                "src/google/protobuf/util/internal/object_writer.cc",
+                "src/google/protobuf/util/internal/proto_writer.cc",
+                "src/google/protobuf/util/internal/protostream_objectsource.cc",
+                "src/google/protobuf/util/internal/protostream_objectwriter.cc",
+                "src/google/protobuf/util/internal/type_info.cc",
+                "src/google/protobuf/util/internal/type_info_test_helper.cc",
+                "src/google/protobuf/util/internal/utility.cc",
+                "src/google/protobuf/util/json_util.cc",
+                "src/google/protobuf/util/message_differencer.cc",
+                "src/google/protobuf/util/time_util.cc",
+                "src/google/protobuf/util/type_resolver_util.cc",
+                "src/google/protobuf/wire_format.cc",
+                "src/google/protobuf/wrappers.pb.cc",
+            }
+        };
+
+        np.IncludeDirectories.Add("src");
+        np.Defines.Add("HAVE_PTHREAD=1");
+        np.CompilerSettings().Add(s => s.WithRTTI(true));
+
+        foreach(var toolchain in toolchains)
+        {
+            var nativeProgramConfiguration = new NativeProgramConfiguration(CodeGen.Release, toolchain, lump: true);
+
+            //we want to build the nativeprogram into an executable. each toolchain can provide different ways of "linking" (static, dynamic, executable are default ones)
+            var format = toolchain.StaticLibraryFormat;
+
+            //and here the magic happens, the nativeprogram gets setup for the specific configuration we just made.
+            np.SetupSpecificConfiguration(nativeProgramConfiguration, format).DeployTo(GetBuildTargetDir(toolchain));
```

Comment:
So as mentioned above you'd do something like this:

    var deployedProgram = np.SetupSpecificConfiguration(nativeProgramConfiguration, format).DeployTo(GetBuildTargetDir(toolchain));
    Backend.Current.AddAliasDependency(toolchain.ActionName, deployedProgram);

And then you can build it like so:

    mono bee.exe macos_x64

Or multiple ones at the same time:

    mono bee.exe macos_x64 macos_arm64